src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -86,138 +89,39 @@ public unsafe int IndexOf(char value, int startIndex, int count)
-        // current character in the string exists in the array of input characters.
-        // There are 256 bits in the map, with each character mapped to 2 bits. Every
-        // character is divided into 2 bytes, and then every byte is mapped to 1 bit.
-        // The character map is an array of 8 integers acting as map blocks. The 3 lsb
-        // in each byte in the character is used to index into this map to get the
-        // right block, the value of the remaining 5 msb are used as the bit position
-        // inside this block.
-        private static unsafe void InitializeProbabilisticMap(uint* charMap, ReadOnlySpan<char> anyOf)
+        public int IndexOfAny(char[] anyOf, int startIndex, int count)
         {
-            bool hasAscii = false;
-            uint* charMapLocal = charMap; // https://github.com/dotnet/runtime/issues/9040
-
-            for (int i = 0; i < anyOf.Length; ++i)
+            if (anyOf is null)
             {
-                int c = anyOf[i];
-
-                // Map low bit
-                SetCharBit(charMapLocal, (byte)c);
-
-                // Map high bit
-                c >>= 8;
-
-                if (c == 0)
-                {
-                    hasAscii = true;
-                }
-                else
-                {
-                    SetCharBit(charMapLocal, (byte)c);
-                }
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.anyOf);
             }
 
-            if (hasAscii)
+            if ((uint)startIndex > (uint)Length)
             {
-                // Common to search for ASCII symbols. Just set the high value once.
-                charMapLocal[0] |= 1u;
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.startIndex, ExceptionResource.ArgumentOutOfRange_Index);
             }
-        }
 
-        private static bool ArrayContains(char searchChar, char[] anyOf)
-        {
-            for (int i = 0; i < anyOf.Length; i++)
+            if ((uint)count > (uint)(Length - startIndex))
             {
-                if (anyOf[i] == searchChar)
-                    return true;
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.count, ExceptionResource.ArgumentOutOfRange_Count);
             }
 
-            return false;
-        }
+            int result = new ReadOnlySpan<char>(ref Unsafe.Add(ref _firstChar, startIndex), count).IndexOfAny(anyOf);
 
-        private static unsafe bool IsCharBitSet(uint* charMap, byte value)
-        {
-            return (charMap[(uint)value & PROBABILISTICMAP_BLOCK_INDEX_MASK] & (1u << (value >> PROBABILISTICMAP_BLOCK_INDEX_SHIFT))) != 0;
-        }
-
-        private static unsafe void SetCharBit(uint* charMap, byte value)
-        {
-            charMap[(uint)value & PROBABILISTICMAP_BLOCK_INDEX_MASK] |= 1u << (value >> PROBABILISTICMAP_BLOCK_INDEX_SHIFT);
+            return result == -1 ? result : result + startIndex;
```

Comment:
```suggestion
            return result < 0 ? result : result + startIndex;
```
Generates a bit better machine code.
(But I don't know if it's actually measurable)

Same on L342.